### PR TITLE
[Multiple Datasource] Fix style of data source option inside popover for data source selector, selectable, multi select components

### PR DESCRIPTION
**Pull request by BionIT**
### Description
This change fixes the style of data source selector/selectable/multi selectable components to hide the test overflow to address the long data source name problem

### Issues Resolved
Fixes https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6430

## Testing the changes
UI should show default icon and truncate the long data source name
1. data source selector
Screenshot from 2024-04-12 21-15-00
2. data source selectable
Screenshot from 2024-04-12 21-13-57
3. multi selectable
Screenshot from 2024-04-12 21-53-34

Function
1. active option provided with empty id string, while hide local cluster
data source selector
        defaultOption={[{id: ''}]}
Screenshot from 2024-04-12 21-55-35

data source selectable
          activeOption: [{id: ''}],
Screenshot from 2024-04-12 21-56-57

2. has default data source
data source selector
Screenshot from 2024-04-12 21-15-00

data source selectable
Screenshot from 2024-04-12 21-14-48

3. empty array passed to data source selector should show placeholder text
Screenshot from 2024-04-12 21-59-41

4. invalid data source id passed in
data source selector
        defaultOption={[{id: 'badid'}]}
Screenshot from 2024-04-12 22-00-53

data source selectable
          activeOption: [{id: 'invalid'}],
Screenshot from 2024-04-12 22-01-47

5. empty string as id and hide local cluster is set to false, should show local cluster
data source selector
        defaultOption={[{id: ''}]}
Screenshot from 2024-04-12 22-03-52

data source selectable
          activeOption: [{id: ''}],
Screenshot from 2024-04-12 22-04-39

6. filter out default data source, hide local cluster is false, should show local cluster
data source selector
Screenshot from 2024-04-12 22-07-09

data source selectable
Screenshot from 2024-04-12 22-07-52

7. pass in valid data source id
data source selector
Screenshot from 2024-04-12 22-09-14

data source selectable
Screenshot from 2024-04-12 22-09-50

8. no available data sources
data source selector
Screenshot from 2024-04-12 22-11-46

data source selectable
Screenshot from 2024-04-12 22-11-59

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by seraphjiang — APPROVED** [review]

File: src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.scss [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+  .dataSourceSelectableOuiFlexGroup {
+    .dataSourceSelectableOuiFlexItem {
+      text-overflow: ellipsis;
+      overflow: hidden;
+      white-space: nowrap;
+      display: inline-block;
+    }
+  }
+}
```
Comment (zhyuanqi) [inline src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.scss:12]: Should we change the name to _data_source_selectable.cscc as a underscore _ shows this is a internal style file

File: src/plugins/data_source_management/public/components/data_source_multi_selectable/data_source_filter_group.tsx [hunk 2]
```diff
@@ -138,7 +139,7 @@ export const DataSourceFilterGroup: React.FC<DataSourceFilterGroupProps> = ({
           data-test-subj="dataSourceMultiSelectFieldSearch"
         />
       </EuiPopoverTitle>
-      <div className="ouiFilterSelect__items" style={{ maxHeight: 400, overflow: 'scroll' }}>
+      <div className="dataSourceFilterGroupItems" style={{}}>
```
Comment (zhyuanqi) [inline src/plugins/data_source_management/public/components/data_source_multi_selectable/data_source_filter_group.tsx:142]: Should we remove style={{}}

Comment (BionIT) [reply src/plugins/data_source_management/public/components/data_source_multi_selectable/data_source_filter_group.tsx:142]: Good catch!

File: src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.scss (hunk 1 above)
Comment (BionIT) [reply src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.scss:12]: We should follow the dev guide `Do not use the underscore _ SASS file naming pattern when importing directly into a javascript file.` in https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/DEVELOPER_GUIDE.md#sass-files

**Review by zhongnansu — APPROVED** [review]

**Review by zhyuanqi — APPROVED** [review]

**Review by ZilongX — APPROVED** [review]

**Pull request merged**